packages/core/src/conflict-resolution.ts

```diff
@@ -389,4 +373,27 @@ export class ConflictResolution {
     const handler = this.handlers.get(initialState.type)
     return this.applyLogToState(handler, resetLog)
   }
+
+  /**
+   * Return `CommitData` with commit and JWS envelope, if applicable and not already present.
+   */
+  private async getCommitData(_commitData: CommitData): Promise<CommitData> {
+    // Clone the `CommitData` so that the Stream state is not affected when commit/JWS data is added to the structure
+    const commitData: CommitData = {
```

Comment:
would `cloneDeep` suffice here?

Reply:
Hah, that was my first thought too. With `cloneDeep`, for some reason, stream state comparison keeps failing in one of the Ceramic tests. I didn't want to hold this PR up to debug but I'm pretty sure it's because the current way populates all the fields, even optional ones (which could be `undefined`) versus cloning, which does not.

I _hated_ leaving it like this but it's about time for this PR to go in so I'll treat this as tech debt and take a look with the next ticket.

Reply:
okay sounds good. Would be nice to understand why cloneDeep breaks things, as I don't think it should, but fine to not hold up this PR over it

Reply:
Also added this one to the maintenance ticket so I don't forget.